tests/Integration/Behaviour/Features/Context/Domain/ShipmentFeatureContext.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Behat\Gherkin\Node\TableNode;
+use PHPUnit\Framework\Assert;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\Query\GetOrderShipments;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\Query\GetShipmentProducts;
+use RuntimeException;
+use Tests\Integration\Behaviour\Features\Context\SharedStorage;
+
+class ShipmentFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @Then the order :orderReference should have the following shipments:
+     *
+     * @param string $orderReference
+     * @param TableNode $table
+     *
+     * @throws RuntimeException
+     */
+    public function verifyOrderShipment(string $orderReference, TableNode $table)
+    {
+        $data = $table->getColumnsHash();
+        $orderId = $this->referenceToId($orderReference);
+        $shipments = $this->getQueryBus()->handle(
+            new GetOrderShipments($orderId)
+        );
+
+        if (count($shipments) === 0) {
+            $msg = 'Order [' . $orderId . '] has no shipments';
+            throw new RuntimeException($msg);
+        }
+
+        foreach ($data as $dataRow) {
+            $carrierId = $this->referenceToId($dataRow['carrier']);
+            $addressId = $this->referenceToId($dataRow['address']);
+            foreach ($shipments as $shipment) {
```

Comment:
I don't think this will work when you have multiple shipments, it's ok for now because there's only one element in the loop But when there will be more it's like you are testing here that all the shipments should match the row from the behat table So once the value start to be different in the rows some errors will popup

You should instead do a single loop that fetches the content from both arrays and compare them, something like

```php
    for ($i = 0; $i < count($data); $i++) {
        $shipmentData = $data[$i];
        $shipment = $shipments[$i];
        // Now compare both
    }
```

Reply:
Actually, something more similar to what you did for products checking bellow, and you should also add a check that the count is equal on both arrays (for products as well)